git/git.go

```diff
@@ -490,13 +490,45 @@ func (c *gitConfig) IsGitVersionAtLeast(ver string) bool {
 	return IsVersionAtLeast(gitver, ver)
 }
 
+// Prepend Git config instructions to disable Git LFS filter
+func PrependConfigToDisableLFS(args ...string) []string {
+	// Before git 2.8, setting filters to blank causes lots of warnings, so use cat instead (slightly slower)
+	// Also pre 2.2 it failed completely. We used to use it anyway in git 2.2-2.7 and
+	// suppress the messages in stderr, but doing that with standard StderrPipe suppresses
+	// the git clone output (git thinks it's not a terminal) and makes it look like it's
+	// not working. You can get around that with https://github.com/kr/pty but that
+	// causes difficult issues with passing through Stdin for login prompts
+	// This way is simpler & more practical.
+	filterOverride := ""
+	if !Config.IsGitVersionAtLeast("2.8.0") {
+		filterOverride = "cat"
+	}
+
+	return append([]string{
+		"-c", fmt.Sprintf("filter.lfs.smudge=%v", filterOverride),
+		"-c", "filter.lfs.clean=",
+		"-c", "filter.lfs.process=",
+		"-c", "filter.lfs.required=false",
+	}, args...)
+}
+
+// Invoke Git with disabled LFS filters
+func Exec(args ...string) *subprocess.Cmd {
```

Comment:
Yes, ideally I'd like that to turn into something like:

```go
h, err := git.HashObject(someReader)
```

Where:

1. The `exec("hash-object", ...)` would live inside of the `git` package, and
2. It would be exposed with a function exported from package `git` called `HashObject()`.